cavern/src/intTest/java/org/opencadc/cavern/QuotaTest.java

```diff
@@ -0,0 +1,200 @@
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+*  $Revision: 4 $
+*
+************************************************************************
+*/
+
+package org.opencadc.cavern;
+
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.security.auth.Subject;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+
+import ca.nrc.cadc.auth.SSLUtil;
+import ca.nrc.cadc.net.OutputStreamWrapper;
+import ca.nrc.cadc.reg.Standards;
+import ca.nrc.cadc.util.FileUtil;
+import ca.nrc.cadc.util.Log4jInit;
+import ca.nrc.cadc.vos.DataNode;
+import ca.nrc.cadc.vos.Direction;
+import ca.nrc.cadc.vos.Protocol;
+import ca.nrc.cadc.vos.Transfer;
+import ca.nrc.cadc.vos.VOS;
+import ca.nrc.cadc.vos.VOSURI;
+import ca.nrc.cadc.vos.client.ClientTransfer;
+import ca.nrc.cadc.vos.client.VOSpaceClient;
+
+/**
+ * Test to verify that deployment and libraries correctly implement authenticated access.
+ *
+ * @author yeunga
+ */
+public class QuotaTest
+{
+    private static Logger log = Logger.getLogger(QuotaTest.class);
+
+    static {
+        Log4jInit.setLevel("ca.nrc.cadc.vos", Level.DEBUG);
+        Log4jInit.setLevel("ca.nrc.cadc.vospace", Level.DEBUG);
+        Log4jInit.setLevel("org.opencadc.cavern", Level.DEBUG);
+    }
+
+    @Test
+    public void testQuotaLimitExceeded()
+    {
+        // test case 1: write a small file, should not exceed quota
+        String testFileName = "src/test/resources/quotaTestFileSmall.txt";
+        writeFile(testFileName, false);
+        
+        // test case 2: write a large file that exceeds the quota
+        testFileName = "src/test/resources/quotaTestFile.txt";
+        writeFile(testFileName, true);
```

Comment:
If this test fails to throw the exception then the file would still be there.  So, the `writeFile` method should probably delete the node at the end.  You could use the VOSpaceClient to do that.

Reply:
Updated quotaTest to delete the test file at the end of the test.